src/useFormStates.ts

```diff
@@ -46,51 +46,50 @@ type UseFormStatesOpts<T, I> = {
 
 export function useFormStates<T, I = T>(opts: UseFormStatesOpts<T, I>): { getFormState: (input: I) => ObjectState<T> } {
   const { config, autoSave, getId, map, addRules } = opts;
+
   const objectStateCache = useMemo<ObjectStateCache<T, I>>(
     () => ({}),
     // Force resetting the cache if config changes
     // eslint-disable-next-line react-hooks/exhaustive-deps
     [config],
   );
   // Keep track of ObjectStates that triggered auto-save when a save was already in progress.
-  const pendingAutoSaves = useRef<ObjectState<T>[]>([]);
-
+  const pendingAutoSaves = useRef<Set<ObjectState<T>>>(new Set());
```

Comment:
Using a `Set` here is a lot similar, with the caveat that a `Set.shift` or `Set.pop` would have been really nice, wtf JS, such bare bones collections...